sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateHiveTableAsSelectCommand.scala

```diff
@@ -21,43 +21,23 @@ import scala.util.control.NonFatal
 
 import org.apache.spark.sql.{Row, SaveMode, SparkSession}
 import org.apache.spark.sql.catalyst.catalog.{CatalogTable, SessionCatalog}
-import org.apache.spark.sql.catalyst.expressions.{Attribute, SortOrder}
 import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
 import org.apache.spark.sql.catalyst.util.CharVarcharUtils
 import org.apache.spark.sql.errors.QueryCompilationErrors
-import org.apache.spark.sql.execution.SparkPlan
-import org.apache.spark.sql.execution.command.{DataWritingCommand, DDLUtils}
-import org.apache.spark.sql.execution.datasources.{HadoopFsRelation, InsertIntoHadoopFsRelationCommand, LogicalRelation, V1WriteCommand, V1WritesUtils}
+import org.apache.spark.sql.execution.command.{DataWritingCommand, DDLUtils, LeafRunnableCommand}
+import org.apache.spark.sql.execution.datasources.{HadoopFsRelation, InsertIntoHadoopFsRelationCommand, LogicalRelation}
 import org.apache.spark.sql.hive.HiveSessionCatalog
 import org.apache.spark.util.Utils
 
-trait CreateHiveTableAsSelectBase extends V1WriteCommand with V1WritesHiveUtils {
+trait CreateHiveTableAsSelectBase extends LeafRunnableCommand {
```

Comment:
do we still need `OptimizedCreateHiveTableAsSelectCommand`? The nested `InsertIntoHadoopFsRelationCommand` should be optimized instead.

Reply:
good question, but if we do not have a `OptimizedCreateHiveTableAsSelectCommand`, how can we get `InsertIntoHadoopFsRelationCommand` ..

The pipeline is:  `CreateHiveTableAsSelectCommand` -> `OptimizedCreateHiveTableAsSelectCommand` -> `InsertIntoHadoopFsRelationCommand`

Reply:
The new pipeline can be `CreateHiveTableAsSelectCommand` -> `hive insertion command` -> `InsertIntoHadoopFsRelationCommand`

Reply:
how to deal with config `spark.sql.hive.convertMetastoreCtas` ? we do not know if the hive insertion is from ctas.

Reply:
or, just deprecated this config

Reply:
remove `OptimizedCreateHiveTableAsSelectCommand` is out of the scope of this pr, how about doing this in a separated pr ?

Reply:
sure